src/test/java/de/tum/in/www1/artemis/service/notifications/NotificationTargetProviderTest.java

```diff
@@ -0,0 +1,185 @@
+package de.tum.in.www1.artemis.service.notifications;
+
+import static de.tum.in.www1.artemis.domain.notification.NotificationTitleTypeConstants.FILE_SUBMISSION_SUCCESSFUL_TITLE;
+import static de.tum.in.www1.artemis.service.notifications.NotificationTargetProvider.*;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.metis.Post;
+import de.tum.in.www1.artemis.domain.notification.Notification;
+import de.tum.in.www1.artemis.domain.notification.NotificationTarget;
+
+public class NotificationTargetProviderTest {
+
+    @Autowired
+    private static NotificationTargetProvider notificationTargetProvider;
+
+    @Mock
```

Comment:
I don't understand why you need mocks for entity objects. I think this should not be done!
Please instead configure the objects normally, i.e. set the id in a new object for the specific test and then use the id.

Reply:
Thank you for pointing this out. I replaced the mocked entity types with real instances and added missing assertThat().as("explanation")... 
I did so with multiple test classes -> that is the reason for the big jump in LoC.